src/libraries/System.Private.CoreLib/src/System/OperatingSystem.cs

```diff
@@ -203,6 +203,22 @@ public static bool IsIOSVersionAtLeast(int major, int minor = 0, int build = 0)
         public static bool IsMacOSVersionAtLeast(int major, int minor = 0, int build = 0)
             => IsMacOS() && IsOSVersionAtLeast(major, minor, build, 0);
 
+        /// <summary>
+        /// Indicates whether the current application is running on Mac Catalyst.
+        /// </summary>
+        public static bool IsMacCatalyst() =>
+#if TARGET_MACCATALYST
+            true;
+#else
+            false;
+#endif
+
+        /// <summary>
+        /// Check for the Mac Catalyst version (returned by 'libobjc.get_operatingSystemVersion') with a >= version comparison. Used to guard APIs that were added in the given Mac Catalyst release.
+        /// </summary>
+        public static bool IsMacCatalystVersionAtLeast(int major, int minor = 0, int build = 0)
+            => IsMacCatalyst() && IsOSVersionAtLeast(major, minor, build, 0);
```

Comment:
Versioning on Mac Catalyst is somewhat complex, because there are two versions in effect: the macOS version the app is running (or targeting), and the equivalent iOS version of the Mac Catalyst API.

Unfortunately Apple decided to use the macOS version in some places, and the iOS version in other places.

For the Xamarin parts of Mac Catalyst (API bindings), we decided to use the iOS version for API version checks (https://github.com/xamarin/xamarin-macios/issues/10453), because this is the version that Apple uses in Objective-C/Swift, and the version they use in their documentation.

Unfortunately it's not the version their version API (`[NSProcessInfo operatingSystemVersionString]`, `[UIDevice systemVersion]`) returns.

The end result is that we had to special-case the API we expose for version checks to handle Mac Catalyst versions (https://github.com/rolfbjarne/xamarin-macios/commit/9314bb5679bb246d7926ba32bb33fd2277da00c0).

I believe `IsMacCatalystVersionAtLeast` should handle versions the same way.

Ref: https://github.com/dotnet/runtime/issues/47768#issuecomment-773585963

Reply:
I'd suggest removing this part from this PR and go over API review approval separately

Reply:
The changes to ref/ were already removed from the PR, and the corresponding changes to SPC are commented out. Should I fully _remove_ those changes, not just comment them?

Reply:
commenting is fine IMHO.